ClickHouse.Client/ADO/ClickHouseConnection.cs

```diff
@@ -42,6 +44,13 @@ public ClickHouseConnection(string connectionString)
         {
             ConnectionString = connectionString;
             var httpClientHandler = new HttpClientHandler() { AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate };
+            
+            if (certificate != null)
+            {
+                httpClientHandler.ClientCertificates.Add(certificate);
+                httpClientHandler.ServerCertificateCustomValidationCallback = (message, certificate2, chain, policy) => true;
+            }
```

Comment:
Disabling server certificate check if a client certificate is provided kind of defeats the point of having TLS